protocols/kad/src/protocol.rs

```diff
@@ -361,6 +413,36 @@ fn resp_msg_to_proto(kad_msg: KadResponseMsg) -> proto::Message {
             }
             msg
         }
+        KadResponseMsg::GetValue {
+            result,
+            closer_peers,
+        } => {
+            let mut msg = proto::Message::new();
+            msg.set_field_type(proto::Message_MessageType::GET_VALUE);
+            msg.set_clusterLevelRaw(9);
+            for peer in closer_peers {
+                msg.mut_closerPeers().push(peer.into());
+            }
+
+            result.map(|(key, value)| {
+                let mut record = proto::Record::new();
+                record.set_key(key.into_bytes());
+                record.set_value(value);
+                msg.set_record(record);
+            });
+
+            msg
+        }
+        KadResponseMsg::PutValue {key, value} =>  {
```

Comment:
See the other comments about `PUT_VALUE` responses. Is this code currently even executed? I don't see a code path that actually _sends_ such a response?

Reply:
No, this is a leftover garbage that needs to go since `PUT_VALUE` doesn't have a response (judging from the docs at least), does it?

Reply:
Interestingly, if I'm reading it correctly, the go implementation [seems to suggest](https://github.com/libp2p/go-libp2p-kad-dht/blob/cf9bd9b649fb7df4a5632357a17d86d5ccc28649/dht.go#L161-L177) that there is indeed a response and that it even contains both `key` and `value` again, which they check whether it matches what they sent.

Reply:
Similarly [for javascript](https://github.com/libp2p/js-libp2p-kad-dht/blob/8d6106b9c9dcf742f1b1985e464c2aa1955464b6/src/private.js#L242-L251). I cannot say that I currently understand why it is done that way.

Reply:
Yes, In the other comment I've pasted a link to handlers in the `handlers.go`, they send out responses for everything but `AddProvider`